src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp

```diff
@@ -829,6 +830,127 @@ ov::pass::NopSliceBeforeGatherElements::NopSliceBeforeGatherElements() {
+    auto optional_stride_const = pattern::optional<ov::op::v0::Constant>();
+    auto pattern = pattern::wrap_type<ov::op::v1::StridedSlice>({input, begin_const, end_const, optional_stride_const});
+
+    ov::matcher_pass_callback matcher_pass_callback = [=](pattern::Matcher& m) {
+        auto node = m.get_match_root();
+        if (node == nullptr) {
+            return false;
+        }
+        auto strided_slice_node = std::dynamic_pointer_cast<ov::op::v1::StridedSlice>(node);
+        // check that all values of the mask is equal 0
+        auto check_mask = [](const std::vector<int64_t>& mask_to_check) {
+            auto it = std::find_if(mask_to_check.begin(), mask_to_check.end(), [](const int64_t& value) {
+                return value != 0;
+            });
+            if (mask_to_check.empty() || it == mask_to_check.end()) {
+                return true;
+            }
+            return false;
+        };
+        // check that we won't do change dimention rank
+        if (!check_mask(strided_slice_node->get_shrink_axis_mask()) ||
+            !check_mask(strided_slice_node->get_new_axis_mask()) ||
+            !check_mask(strided_slice_node->get_ellipsis_mask())) {
+            return false;
+        }
+        // check that that we will take all values
+        if (node->get_input_size() == 4 && !op::util::is_constant_and_all_values_equal_int(node->input_value(3), 1)) {
+            return false;
+        }
+        auto begin_node = strided_slice_node->get_input_node_shared_ptr(1);
+        if (const auto& begin_constant_node = ov::util::get_constant_from_source(begin_node)) {
+            const auto& values = begin_constant_node->cast_vector<int64_t>();
+            auto begin_mask = strided_slice_node->get_begin_mask();
+            const auto dim_size = std::min(begin_mask.size(), values.size());
+            for (size_t i = 0; i < dim_size; ++i) {
+                // if mask == 1 then ignore the begin_mask_value else check
+                // if values[i] == 0 then take whole tensor else take part of a tensor
+                if (!begin_mask[i] && values[i]) {
+                    return false;
+                }
+            }
+        } else {
+            return false;
+        }
+
+        auto end_node = strided_slice_node->get_input_node_shared_ptr(2);
+        if (const auto& end_constant_node = ov::util::get_constant_from_source(end_node)) {
+            const auto& values = end_constant_node->cast_vector<int64_t>();
+            auto end_mask = strided_slice_node->get_end_mask();
+            const auto dim_size = std::min(end_mask.size(), values.size());
+            for (size_t i = 0; i < dim_size; ++i) {
+                // if mask == 1 then ignore the begin_mask_value else check
+                // if values[i] == max then take whole tensor else take part of a tensor
+                if (!end_mask[i] && values[i] != std::numeric_limits<int64_t>::max()) {
```

Comment:
can we check "end" input precision and take max based on that?
not so sure if we have any helper function for that 

I hope this will work
```
auto el_type = end->get_element_type();
if (el_type == element::i64) {
    std::numeric_limits<fundamental_type_for<element::i64>>::max();
} else if (...) {
}
```